frontend/src/container/NewDashboard/DashboardVariablesSelection/index.tsx

```diff
@@ -38,16 +47,23 @@ function DashboardVariableSelection({
 		const updatedVariablesData = { ...variables };
 		updatedVariablesData[name].allSelected = value;
 		updateDashboardVariables(updatedVariablesData);
+		onVarChanged(name);
 	};
 
 	return (
 		<Row style={{ gap: '1rem' }}>
 			{map(sortBy(Object.keys(variables)), (variableName) => (
 				<VariableItem
 					key={`${variableName}${variables[variableName].modificationUUID}`}
-					variableData={{ name: variableName, ...variables[variableName] }}
+					existingVariables={variables}
+					variableData={{
```

Comment:
can wrap under memo

Reply:
Why is this recommended here?

Reply:
This needed to be in line because the values from map data are being used so it can't be moved outside.